api/helpers/find-or-create-current-match.js

```diff
@@ -0,0 +1,70 @@
+const dayjs = require('dayjs');
+
+module.exports = {
+  friendlyName: 'Find Or Create Current Match Between Two Players',
+
+  description:
+    'Finds the relevant match for the specified players for the current week, and creates the match if it does not already exist',
+
+  inputs: {
+    player1: {
+      type: 'number',
+      descritiption: 'The database id of the first player in the desired match',
+      example: 44,
+      required: true,
+    },
+    player2: {
+      type: 'number',
+      descritiption: 'The database id of the second player in the desired match',
+      example: 45,
+      required: true,
+    },
+  },
+
+  fn: async ({ player1, player2 }, exits) => {
+    const currentTime = dayjs();
+    try {
+      const currentSeason = await Season.findOne({
+        startTime: { '<=': currentTime.valueOf() },
+        endTime: { '>=': currentTime.valueOf() },
+      });
+      // FIXME: Handle missing season gracefully
+      if (!currentSeason) {
+        return exits.success(null);
+      }
+      // Find relevant match between specified players for current week
+      const seasonStartTime = dayjs(currentSeason.startTime);
+      const weeksSinceSeasonStart = currentTime.diff(seasonStartTime, 'week');
+      const currentWeekStartTime = seasonStartTime.add(weeksSinceSeasonStart, 'week').valueOf();
+      const currentWeekEndTime = seasonStartTime.add(weeksSinceSeasonStart + 1, 'week').valueOf();
```

Comment:
This feels like maybe it should be a utility that can be tested functionally.

Reply:
Yes I agree, but I think it needs to be done against a booted backend because it's query for data in a database (or in mocked-in-memory sails-disk), so it's not a pure function